File: src/main/java/com/mbayou/kick4k/categories/Category.java
package com.mbayou.kick4k.categories;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Category {
    private final Integer id;
    private final String name;
    private final String thumbnail;

    @JsonCreator
    public Category(@JsonProperty("id") Integer id, @JsonProperty("name") String name, @JsonProperty("thumbnail") String thumbnail) {
        this.id = id;
        this.name = name;
        this.thumbnail = thumbnail;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getThumbnail() {
        return thumbnail;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/categories/Category.kt
package com.mbayou.kick4k.categories

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class Category @JsonCreator constructor(
    @JsonProperty("id") val id: Int?,
    @JsonProperty("name") val name: String?,
    @JsonProperty("thumbnail") val thumbnail: String?,
)
